expression/builtin_compare.go

```diff
@@ -294,11 +296,81 @@ func isTemporalColumn(expr Expression) bool {
 	return true
 }
 
+func tryToConvertConstantInt(con *Constant, ctx context.Context) *Constant {
+	if con.GetTypeClass() == types.ClassInt {
+		return con
+	}
+	sc := ctx.GetSessionVars().StmtCtx
+	i64, err := con.Value.ToInt64(sc)
+	if err != nil {
+		return con
+	}
+	return &Constant{
+		Value:   types.NewIntDatum(i64),
+		RetType: types.NewFieldType(mysql.TypeLonglong),
+	}
+}
+
+func refineTwoArgs(con *Constant, op opcode.Op, ctx context.Context) *Constant {
```

Comment:
1. comment for this func
2. why not make this func be member method of compareFunctionClass.